factory-package-news/factory-package-news-web.py

```diff
@@ -26,16 +26,21 @@
 import re
 import os
 import sys
+from urlparse import urlparse
 
 import re
 digits_re = re.compile('^[0-9]+$')
 
-_dir = "/var/lib/openqa/factory/snapshot-changes"
+BASE_DIR = '/var/lib'
 
 app = Flask(__name__)
 
+def get_dir(url):
+    return os.path.join(BASE_DIR, urlparse(url).path.lstrip('/'))
```

Comment:
For kicks I could check that the resulting path is is within the expected directory or could split it for the two parts and ensure only letters. Seems like this is unnecessary since the paths have to match literal strings in our config.